api/internal/utils/timedcall_test.go

```diff
@@ -62,3 +63,15 @@ func TestTimedCallSlowWithError(t *testing.T) {
 		t.Fail()
 	}
 }
+
+func TestTimedCallGoroutineLeak(t *testing.T) {
+	beforeGroutine := runtime.NumGoroutine()
+	_ = TimedCall("expect no goroutine leaks", time.Second, func() error {
+		time.Sleep(2 * time.Second)
+		return fmt.Errorf("not done")
+	})
```

Comment:
could you be more specify on what has not done? it would help a lot. Thanks

Reply:
Sorry for the misrepresentation here.
`not done` I would say would cause the goroutine to fail to exit and not finish.
Would it be better to change it to `return fmt.Error("function done")`?

Reply:
I‘m not buying just return nil, I think something like ``goroutine to fail to exit and not finish`` or ``function done`` is fine

Reply:
Thanks for the suggestion, the code has been changed to `return fmt.Errorf("function done")`